crates/cheatcodes/src/test.rs

```diff
@@ -64,28 +54,17 @@ impl Cheatcode for zkRegisterContractCall {
             zkDeployedBytecode,
         } = self;
 
-        let new_contract = DualCompiledContract {
-            name: name.clone(),
-            zk_bytecode_hash: zkBytecodeHash.0.into(),
-            zk_deployed_bytecode: zkDeployedBytecode.to_vec(),
-            //TODO: add argument to cheatcode
-            zk_factory_deps: vec![],
-            evm_bytecode_hash: *evmBytecodeHash,
-            evm_deployed_bytecode: evmDeployedBytecode.to_vec(),
-            evm_bytecode: evmBytecode.to_vec(),
-        };
-
-        if let Some(existing) = ccx.state.dual_compiled_contracts.iter().find(|contract| {
-            contract.evm_bytecode_hash == new_contract.evm_bytecode_hash &&
-                contract.zk_bytecode_hash == new_contract.zk_bytecode_hash
-        }) {
-            warn!(name = existing.name, "contract already exists with the given bytecode hashes");
-            return Ok(Default::default())
-        }
-
-        ccx.state.dual_compiled_contracts.push(new_contract);
-
-        Ok(Default::default())
+        ccx.with_strategy(|strategy, _ccx| {
```

Comment:
Maybe the Ext trait gets simplified if we just implement zksync cheatcodes inline?